plugin/src/main/java/org/opensearch/ml/action/tasks/GetTaskTransportAction.java

```diff
@@ -84,6 +141,128 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLTask
+                        connectorAccessControlHelper
+                            .getConnector(client, model.getConnectorId(), ActionListener.runBefore(listener, threadContext::restore));
+                    }
+                } else {
+                    actionListener.onFailure(new ResourceNotFoundException("Can't find connector " + model.getConnectorId()));
+                }
+            }, e -> {
+                log.error("Failed to retrieve the ML model with the given ID", e);
+                actionListener.onFailure(e);
+            }));
+        } catch (Exception e) {
+            // fetch the connector
+            log.error("Unable to fetch status for ml task ", e);
+        }
+    }
+
+    private void executeConnector(
+        Connector connector,
+        MLInput mlInput,
+        String taskId,
+        MLTask mlTask,
+        Map<String, Object> transformJob,
+        ActionListener<MLTaskGetResponse> actionListener
+    ) {
+        connector.decrypt(BATCH_PREDICT_STATUS.name(), (credential) -> encryptor.decrypt(credential));
+        RemoteConnectorExecutor connectorExecutor = MLEngineClassLoader.initInstance(connector.getProtocol(), connector, Connector.class);
+        connectorExecutor.setScriptService(scriptService);
+        connectorExecutor.setClusterService(clusterService);
+        connectorExecutor.setClient(client);
+        connectorExecutor.setXContentRegistry(xContentRegistry);
+        connectorExecutor.executeAction(BATCH_PREDICT_STATUS.name(), mlInput, ActionListener.wrap(taskResponse -> {
+            processTaskResponse(mlTask, taskId, taskResponse, transformJob, actionListener);
+        }, e -> { actionListener.onFailure(e); }));
+    }
 
+    private void processTaskResponse(
+        MLTask mlTask,
+        String taskId,
+        MLTaskResponse taskResponse,
+        Map<String, Object> remoteJob,
+        ActionListener<MLTaskGetResponse> actionListener
+    ) {
+        try {
+            ModelTensorOutput tensorOutput = (ModelTensorOutput) taskResponse.getOutput();
+            if (tensorOutput != null && tensorOutput.getMlModelOutputs() != null && !tensorOutput.getMlModelOutputs().isEmpty()) {
+                ModelTensors modelOutput = tensorOutput.getMlModelOutputs().get(0);
+                if (modelOutput.getMlModelTensors() != null && !modelOutput.getMlModelTensors().isEmpty()) {
+                    Map<String, Object> remoteJobStatus = (Map<String, Object>) modelOutput.getMlModelTensors().get(0).getDataAsMap();
+                    if (remoteJobStatus != null) {
+                        remoteJob.putAll(remoteJobStatus);
+                        Map<String, Object> updatedTask = new HashMap<>();
+                        updatedTask.put(REMOTE_JOB_FIELD, remoteJob);
+
+                        if ((remoteJob.containsKey("status") && remoteJob.get("status").equals("completed"))
```

Comment:
Is it possible that the key-value pair in remoteJobStatus from remote service is not "status" -> "completed" || "Completed", but some other names or status?

Reply:
Yeah it can be other values like in progress, cancelled, etc